includes/classes/traits/NotifierManager.php

```diff
@@ -94,6 +94,78 @@ function notify($eventID, $param1 = array(), &$param2 = null, &$param3 = null, &
         }
     }
 
+    /**
+     * Requires a define page, and calls regisistered Observers for content output.
+     *
+     * @param [type] $eventID
+     * @param array $param1
+     * @param [type] ...$params
+     * @return void
+     */
+    function insertContent($eventID, $param1 = array(), &...$params)
+    {
+        global $l;
+
+        // Output any 'define_' page, e.g. CONTENT_FOO will output define_foo.php
+        $define_page_name = strtolower(str_replace('CONTENT_', 'DEFINE_', $eventID));
+        $this->outputDefinePage($define_page_name);
+
+        $observers = $this->getRegisteredObservers();
+        if (empty($observers)) {
+            return;
+        }
+        foreach ($observers as $key => $obs) {
+            // Skip observer if it's not registered for a CONTENT_ event.
+            if (!str_starts_with($obs['eventID'], 'CONTENT_')) {
+                continue;
+            }
+
+            // If there is a mapping provided on the observer from event name to define page, use it and continue.
+            if (!empty($obs['obs']->contentDefinePageMap) && array_key_exists($eventID, $obs['obs']->contentDefinePageMap)) {
+                $this->outputDefinePage($obs['obs']->contentDefinePageMap[$eventID]);
+                continue;
+            }
+
+            // Notify the listening observer that this event has been triggered
+            $snake_case_method = strtolower($eventID);
+            if (method_exists($obs['obs'], $snake_case_method)) {
+                $obs['obs']->{$snake_case_method}($this, $eventID, $params);
+            } else {
+                // If no update handler method exists then trigger an error so the problem is logged
```

Comment:
This `else` clause might be skippable, since it's guarding against older/legacy observer syntax support.
So if what's being built here won't ever need that legacy support, maybe the clause isn't needed.
Not that it's harmful to keep either.

Reply:
I thought this was still useful.  If an observer registers interest in `CONTENT_FOO` and then does not declare a handler function, that's a bug that needs raising.  Note that if the `contentDefinePageMap` contains a mapping, the loop is `continue`d and the error would not be raised, as intended.

Now I think about it, my 'sniff for define page' logic is poor implementation of the design, you still need an observer registered for `CONTENT_FOO` in order to get `define_foo.php` output by a hook.  If the design intent is that having a `define_foo.php` on disk would get it output at the `CONTENT_FOO` hook point, then one should not have to create an observer at all.  I will need to fix this.